x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -234,6 +239,14 @@ export const createLifecycleExecutor =
       result.forEach((hit) => {
         const alertInstanceId = hit._source ? hit._source[ALERT_INSTANCE_ID] : void 0;
         if (alertInstanceId && hit._source) {
+          const alertState = hit._source[ALERT_STATUS];
```

Comment:
Should we be doing this inside the alerting framework alerts client as well?

Reply:
I haven't found anywhere that it's necessary. This is just to get the Observability event log to behave the way we're expecting, so that an alert that gets re-enabled starts generating new alert events instead of setting existing Untracked event logs to Active.

Reply:
We have the index threshold rule and the ES query rule writing alert docs now too and I believe that their untracked alerts will be reset to active instead of becoming a new alert when a rule is re enabled

Reply:
@Zacqary Ying and I talked about that and we think that we can do something simple like that https://gist.github.com/XavierM/ac5e2cfb4d15d6de23e1a9762d9d884a

what do you think?

Reply:
@XavierM That seems to work. Pushing it now.